frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import * as _ from 'lodash';
+import {
+  Alert,
+  DataList,
+  DataListCheck,
+  DataListItem,
+  DataListItemRow,
+  DataListCell,
+  DataListItemCells,
+} from '@patternfly/react-core';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { setTableColumns } from '../../actions/ui';
+
+export const MAX_VIEW_COLS = 9;
+
+export const NAME_COLUMN_ID = 'NAME';
+const READ_ONLY_COLS = [NAME_COLUMN_ID];
+
+const DataListRow: React.FC<DataListRowProps> = ({ column, onChange, disableUncheckedRow }) => (
+  <DataListItem aria-labelledby={`table-column-management-item-${column.ID}`} key={column.ID}>
+    <DataListItemRow>
+      <DataListCheck
+        isDisabled={(disableUncheckedRow && !column.visible) || READ_ONLY_COLS.includes(column.ID)}
+        aria-labelledby={`table-column-management-item-${column.ID}`}
+        checked={column.visible}
+        name={column.title}
+        id={column.ID}
+        onChange={onChange}
+      />
+      <DataListItemCells
+        dataListCells={[
+          <DataListCell id={`table-column-management-item-${column.ID}`} key={column.ID}>
+            {column.title}
+          </DataListCell>,
+        ]}
+      />
+    </DataListItemRow>
+  </DataListItem>
+);
+
+export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
+  cancel,
+  close,
+  columnManagementID,
+  columnManagementType,
+  selectedColumns,
+}) => {
+  const dispatch = useDispatch();
+  const initialDefaultColumns = !_.isEmpty(selectedColumns)
+    ? selectedColumns.filter((column) => column.title.length > 0 && !column.additional)
```

Comment:
```suggestion
    ? selectedColumns.filter((column) => column.title && !column.additional)
```

Reply:
@spadgett needed this syntax for kebab columns before your last review, however, since we've added id's, i can make this more clear by checking for presence of column.id now